src/iuse_actor.cpp

```diff
@@ -2133,21 +2160,91 @@ bool repair_item_actor::can_repair( player &pl, const item &tool, const item &fi
         return false;
     }
 
-    if( fix.damage == 0 && fix.has_flag("PRIMITIVE_RANGED_WEAPON") ) {
+    if( fix.has_flag("VARSIZE") && !fix.has_flag("FIT") ) {
+        return true;
+    }
+
+    if( fix.damage > 0 ) {
+        return true;
+    }
+
+    if( fix.damage < 0 ) {
+        if( print_msg ) {
+            pl.add_msg_if_player( m_info, _("Your %s is already enhanced."), fix.tname().c_str() );
+        }
+        return false;
+    }
+
+    if( fix.has_flag("PRIMITIVE_RANGED_WEAPON") ) {
         if( print_msg ) {
             pl.add_msg_if_player( m_info, _("You cannot improve your %s any more this way."), fix.tname().c_str());
         }
         return false;
     }
 
-    if( fix.damage >= 0 || (fix.has_flag("VARSIZE") && !fix.has_flag("FIT")) ) {
-        return true;
+    return true;
+}
+
+std::pair<float, float> repair_item_actor::repair_chance(
+    const player &pl, const item &fix, repair_item_actor::repair_type action_type ) const
+{
+    ///\EFFECT_TAILOR randomly improves clothing repair efforts
+    ///\EFFECT_MECHANICS randomly improves metal repair efforts
+    const int skill = pl.get_skill_level( used_skill );
+    const int recipe_difficulty = repair_recipe_difficulty( pl, fix );
+    int action_difficulty = 0;
+    switch( action_type ) {
+        case RT_REPAIR:
+            action_difficulty = fix.damage;
+            break;
+        case RT_REFIT:
+            // Let's make refitting as hard as recovering an almost-wrecked item
+            action_difficulty = MAX_ITEM_DAMAGE;
+            break;
+        case RT_REINFORCE:
+            // Reinforcing is at least as hard as refitting
+            action_difficulty = std::max( MAX_ITEM_DAMAGE, recipe_difficulty );
+            break;
+        default:
+            std::make_pair( 0.0f, 0.0f );
```

Comment:
You're constructing an anonymous instance of `std::pair`, but don't do anything with it. Is this supposed to be returned?